### PR TITLE
Maint/159 clean up

### PR DESCRIPTION
**Pull request by chenqi0805**
*Issue #, if available:*
#159 

*Description of changes:*
This PR

- refactors apmtracesource into otel-trace-source and otel-trace-raw-processor
- fix other places that depends on apmtracesource
- renames upper-case to string_converter for StringProcessor and exposes the boolean to switch between upper and lower case

Note: tested on sample app example

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: situp-plugins/common/src/main/java/com/amazon/situp/plugins/processor/StringProcessor.java [hunk 1]
```diff
@@ -25,19 +27,16 @@
...
+        this.toUpperCase = pluginSetting.getBooleanOrDefault("upper_case", true);
     }
 
     @Override
     public Collection<Record<String>> execute(final Collection<Record<String>> records) {
         final Collection<Record<String>> modifiedRecords = new ArrayList<>(records.size());
         for (Record<String> record : records) {
             final String recordData = record.getData();
-            modifiedRecords.add(new Record<>(recordData.toUpperCase()));
+            final String newData = toUpperCase? recordData.toUpperCase() : recordData.toLowerCase();
```
Comment (yadavcbala) [inline situp-plugins/common/src/main/java/com/amazon/situp/plugins/processor/StringProcessor.java:38]: Nit: toUpperCase<space>?

Comment (yadavcbala) [reply situp-plugins/common/src/main/java/com/amazon/situp/plugins/processor/StringProcessor.java:38]: Also, if upper_case is not set - is it needed to do lower_case ? [I don't have a problem but if we expand this processor with other attributes and I don't want case modification at all - it is not possible right;]

Comment (chenqi0805) [reply situp-plugins/common/src/main/java/com/amazon/situp/plugins/processor/StringProcessor.java:38]: change variable name to `upperCase`?

File: situp-plugins/otel-trace-raw-processor/README.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# OTel Trace Raw Processor
+
+This is a processor that serializes collection of `ExportTraceServiceRequest` sent from [otel-trace-source](../situp-plugins/otel-trace-source) into collection of string records. 
+
+```
+processor:
+    otel_trace_source:
```
Comment (yadavcbala) [inline situp-plugins/otel-trace-raw-processor/README.md:7]:
```
processor:
    - otel_trace_source:
```
?

File: situp-plugins/otel-trace-raw-processor/src/test/java/com.amazon.situp.plugins.processor.oteltrace/OTelTraceRawProcessorTest.java [hunk 3]
```diff
@@ -44,6 +45,6 @@ public void testExportRequest() throws InvalidProtocolBufferException, JsonProce
         JsonFormat.parser().merge(sampleRequest, builder);
         final ExportTraceServiceRequest exportTraceServiceRequest = builder.build();
         final List<Record<String>> processedRecords = (List<Record<String>>) oTelTraceRawProcessor.execute(Collections.singletonList(new Record<>(exportTraceServiceRequest)));
-        assertThat(processedRecords.size()).isEqualTo(6);
+        Assertions.assertThat(processedRecords.size()).isEqualTo(6);
```
Comment (yadavcbala) [inline situp-plugins/otel-trace-raw-processor/src/test/java/com.amazon.situp.plugins.processor.oteltrace/OTelTraceRawProcessorTest.java:48]: nit: thought we were considering to use hamcrest over assertj; I am not advocating one over the other but just a question

File: situp-plugins/common/src/main/java/com/amazon/situp/plugins/processor/StringProcessor.java (hunk 1 above)
Comment (yadavcbala) [reply situp-plugins/common/src/main/java/com/amazon/situp/plugins/processor/StringProcessor.java]: say if we add other attributes (features) later - we are doing either upper case or lower case -> there is no way for me to use this processor without modifying the case; should we not do `recordData.toLowerCase();` if `toUppercase is false`. What do u think ?

Comment (chenqi0805) [reply situp-plugins/common/src/main/java/com/amazon/situp/plugins/processor/StringProcessor.java]: > Also, if upper_case is not set - is it needed to do lower_case ? [I don't have a problem but if we expand this processor with other attributes and I don't want case modification at all - it is not possible right;]

For the item laid out in #159 , this is the desired behavior. @kowshikn Shall we leave more customized string conversion to future work?

File: situp-plugins/otel-trace-raw-processor/src/test/java/com.amazon.situp.plugins.processor.oteltrace/OTelTraceRawProcessorTest.java (hunk 3 above)
Comment (chenqi0805) [reply situp-plugins/otel-trace-raw-processor/src/test/java/com.amazon.situp.plugins.processor.oteltrace/OTelTraceRawProcessorTest.java:48]: Ah. Looks like our plugins are still mostly using assertj. So maybe we leave it for separate PR?

File: situp-plugins/common/src/main/java/com/amazon/situp/plugins/processor/StringProcessor.java (hunk 1 above)
Comment (yadavcbala) [reply situp-plugins/common/src/main/java/com/amazon/situp/plugins/processor/StringProcessor.java]: I am fine with it for now - but we renamed it to String_converter when its basically case converter - but its fine if we later are going to add features to this - in that case I think we should leave an issue for future

File: situp-plugins/otel-trace-raw-processor/src/test/java/com.amazon.situp.plugins.processor.oteltrace/OTelTraceRawProcessorTest.java (hunk 3 above)
Comment (yadavcbala) [reply situp-plugins/otel-trace-raw-processor/src/test/java/com.amazon.situp.plugins.processor.oteltrace/OTelTraceRawProcessorTest.java:48]: makes sense

**Comment by kowshikn** [comment]
Could you resolve the conflicts?

File: situp-plugins/common/src/main/java/com/amazon/situp/plugins/processor/StringProcessor.java [hunk 4]
```diff
@@ -13,9 +13,11 @@
  * An simple String implementation of {@link Processor} which generates new Records with upper case content. The current
  * simpler implementation does not handle errors (if any).
```
Comment (wrijeff) [inline situp-plugins/common/src/main/java/com/amazon/situp/plugins/processor/StringProcessor.java:14]: Docs are no longer in sync - can now lowercase strings depending on configuration.

File: situp-plugins/common/src/main/java/com/amazon/situp/plugins/processor/StringProcessor.java [hunk 5]
```diff
@@ -13,9 +13,11 @@
  * An simple String implementation of {@link Processor} which generates new Records with upper case content. The current
```
Comment (kowshikn) [inline situp-plugins/common/src/main/java/com/amazon/situp/plugins/processor/StringProcessor.java:13]: nit: Write a readme for this to keep it consistent with rest of the plugins.

**Comment by kowshikn** [comment]
Create a Readme in the plugins::common subproject and cover the below plugins.
1. Bounded blocking
2. String converter
3. File sink
4. File source
5. Std source 
6. std sink

File: situp-plugins/common/src/main/java/com/amazon/situp/plugins/processor/StringProcessor.java (hunk 4 above)
Comment (chenqi0805) [reply situp-plugins/common/src/main/java/com/amazon/situp/plugins/processor/StringProcessor.java:14]: Good catch!

**Comment by chenqi0805** [comment]
> Create a Readme in the plugins::common subproject and cover the below plugins.

Discussed offline and will do in a separate PR.

File: situp-plugins/common/src/main/java/com/amazon/situp/plugins/processor/StringProcessor.java (hunk 1 above)
Comment (chenqi0805) [reply situp-plugins/common/src/main/java/com/amazon/situp/plugins/processor/StringProcessor.java]: Discussed with Kowshik offline, for now it is only a sample processor. Will customize it later.

**Comment by chenqi0805** [comment]
Also updated overview.md. Will check running the `sample configuration`

**Review by yadavcbala — APPROVED** [review]

**Review by wrijeff — APPROVED** [review]

**Comment by chenqi0805** [comment]
@kowshikn Made some changes to overview.md and verified all the sample configurations work as expected.

**Review by kowshikn — APPROVED** [review]

File: situp-plugins/common/src/main/java/com/amazon/situp/plugins/processor/StringProcessor.java (hunk 5 above)
Comment (chenqi0805) [reply situp-plugins/common/src/main/java/com/amazon/situp/plugins/processor/StringProcessor.java]: Done

**Pull request merged**